orchestrator/main_loops.go

```diff
@@ -280,15 +281,34 @@ func (p *peggyOrchestrator) CheckFeeThreshold(
 		return true
 	}
 
+	decimals, err := p.peggyContract.GetERC20Decimals(ctx, erc20Contract, p.peggyContract.FromAddress())
+	if err != nil {
+		p.logger.Err(err).Str("token_contract", erc20Contract.String()).Msg("failed to get token decimals")
+		return false
+	}
+
+	p.logger.Debug().
+		Uint8("decimals", decimals).
+		Str("token_contract", erc20Contract.String()).
+		Msg("got token decimals")
+
 	tokenPriceInUSD, err := p.priceFeeder.QueryUSDPrice(erc20Contract)
 	if err != nil {
 		return false
 	}
 
 	tokenPriceInUSDDec := decimal.NewFromFloat(tokenPriceInUSD)
-	totalFeeInUSDDec := decimal.NewFromBigInt(totalFee.BigInt(), -18).Mul(tokenPriceInUSDDec)
+	totalFeeInUSDDec := decimal.NewFromBigInt(totalFee.BigInt(), -int32(decimals)).Mul(tokenPriceInUSDDec)
 	minFeeInUSDDec := decimal.NewFromFloat(minFeeInUSD)
 
+	p.logger.Debug().
+		Str("token_contract", erc20Contract.String()).
+		Float64("token_price_in_usd", tokenPriceInUSD).
+		Int64("total_fees", totalFee.Int64()).
+		Float64("total_fee_in_usd", totalFeeInUSDDec.InexactFloat64()).
+		Float64("min_fee_in_usd", minFeeInUSDDec.InexactFloat64()).
+		Msg("checking if token fees meets set threshold amount")
```

Comment:
```suggestion
		Msg("token fees meet minimum batch fee threshold")
```

Reply:
This is not necessarily true, the check is performed in the return

Reply:
You're right!